libs/Microsoft.MixedReality.WebRTC.Native/test/simple_interop.cpp

```diff
@@ -22,14 +22,17 @@ void SimpleInterop::Unregister(mrsPeerConnectionHandle pc) noexcept {
 
 void* SimpleInterop::CreateObject(ObjectType type) noexcept {
   const uint32_t id = free_id_.fetch_add(1);
-  auto handle = std::make_unique<Handle>(Handle{this, type, id});
-  void* interop_handle = handle.get();
   try {
+    auto handle = std::make_unique<Handle>(Handle{this, type, id});
+    void* interop_handle = handle.get();
     auto lock = std::scoped_lock{objects_map_mutex_};
     objects_map_.emplace(interop_handle, std::move(handle));
+    return interop_handle;
   } catch (...) {
+    // Make sure the test fails
+    []() { ASSERT_TRUE(false); }();
```

Comment:
nit: why the lambda? and why not at line 51?

Reply:
The Google Test macros can only be used inside a function that returns `void` unfortunately. And I don't know of any other way to force-fail a test. So wrapping inside a lambda, and calling the lambda immediately. I can also write a `void ForceTestFailure()` helper, is that better for clarity?

Reply:
Might be worth it for readability but it's not super-important.

Reply:
Fixed. That's much clearer.